op-challenger/game/fault/trace/vm/executor.go

```diff
@@ -33,22 +33,28 @@ type Config struct {
 	InfoFreq         uint // Frequency of progress log messages (in VM instructions)
 }
 
+type VmArgs interface {
+	Cfg() Config
+	LocalInputs() *utils.LocalGameInputs
+	SetLocalInputs(utils.LocalGameInputs)
+	FillHostCommand(args *[]string, dataDir string) error
```

Comment:
This is more efficient though, no? We're passing a mutable reference down, rather than passing by value and copying the `args`, just to reproduce another instance of `args` via `append`, which would re-allocate out of place due to the out-of-capacity extension.

Reply:
Who cares? :) We're about to spend 3 hours executing the VM and we've never executed this in production. I don't think we need to compromise the design to avoid copying a few strings.  We'll wind up needing to increase the size of the `args` array anyway which means copying data because we don't create it with a sufficiently large size (we don't know what size it needs to be upfront).